frontend/packages/dev-console/src/components/import/deployImage-validation-utils.ts

```diff
@@ -27,13 +28,21 @@ export const deployValidationSchema = yup.object().shape({
       then: yup.object({
         minpods: yup
           .number()
-          .integer('Min Pods must be an Integer.')
-          .min(0, 'Min Pods must be greater than or equal to 0.'),
+          .test(isInteger('Min Pods must be an Integer.'))
+          .min(0, 'Min Pods must be greater than or equal to 0.')
+          .max(
+            Number.MAX_SAFE_INTEGER,
+            `Min Pods must be lesser than or equal to ${Number.MAX_SAFE_INTEGER}`,
+          ),
         maxpods: yup
           .number()
           .transform((cv) => (_.isNaN(cv) ? undefined : cv))
-          .integer('Max Pods must be an Integer.')
+          .test(isInteger('Max Pods must be an Integer.'))
```

Comment:
Why do you need a separate util `isInteger` here? What is the issue with using `.integer` from `yup`?

Reply:
In yup, to validate an Integer they have used the below validation code
`value == null || (value | 0) == value`
The maximum value it can validate successful is 2147483647 anything more than that, it fails.
`
>let a = 2147483647;
>a|0
<2147483647
>(a+1)|0
<-2147483648
>(a+2)|0
<-2147483647
`
This bug is fixed in https://github.com/jquense/yup/pull/405 but they have not made any release with this fix. Myself and @christianvogt had a discussion and we conclude our own way of fixing until the formal release is made.

You can track the conversation in https://jira.coreos.com/browse/ODC-1225

Reply:
Also the restriction of MAX_SAFE_INTEGER to max value is to enable correct validation.

`>let x = Number.MAX_SAFE_INTEGER + 1
<undefined
>let y = Number.MAX_SAFE_INTEGER + 2
<undefined
>x
<9007199254740992
>y
<9007199254740992
>x === y
<true`

Reply:
Got it. Thanks.